### PR TITLE
Fix doc build

### LINKED ISSUE #124: docs.rs doc build broken
I noticed that https://docs.rs/async-executor/1.13.0 has a broken build. The docs.rs build logs say:

```rust
[INFO] [stderr] error[E0658]: `#[doc(cfg)]` is experimental
[INFO] [stderr]   --> src/lib.rs:61:20
[INFO] [stderr]    |
[INFO] [stderr] 61 | #[cfg_attr(docsrs, doc(cfg(any(feature = "static"))))]
[INFO] [stderr]    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[INFO] [stderr]    |
[INFO] [stderr]    = note: see issue #43781 <https://github.com/rust-lang/rust/issues/43781> for more information
[INFO] [stderr]    = help: add `#![feature(doc_cfg)]` to the crate attributes to enable
```

It appears that the problem is that #122 added a use of `#[doc(cfg(...))]` without also adding `#![feature(doc_cfg)]`.

### PR DESCRIPTION
**Pull request by linyihai**
Hello，I made this PR to address https://github.com/smol-rs/async-executor/issues/124

Closes #124

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@
     html_logo_url = "https://raw.githubusercontent.com/smol-rs/smol/master/assets/images/logo_fullsize_transparent.png"
 )]
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![feature(doc_cfg)]
```
Comment (taiki-e) [inline src/lib.rs:41]:
```suggestion
#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
```

**Review by taiki-e — CHANGES_REQUESTED** [review]
Thanks for the PR. Please move unrelated changes into a separate PR.

File: src/lib.rs (hunk 1 above)
Comment (linyihai) [reply src/lib.rs]: I didn't think it could be used together. It has been modified.

**Comment by linyihai** [comment]
Em, the clippy warning can be fixed in my next PR together,  It seems to be just a matter of format.

**Review by taiki-e — APPROVED** [review]

**Pull request merged**